### PR TITLE
Messages modal converted to Bootstrap

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Move modals from mootools to Bootstrap

this now serves only the styling and the image required in Hathor. 

To test this you should go to: https://github.com/joomla-extensions/messages/pull/7

Create an installable package and then apply also this PR!

### CODE REVIEW

**Comment by brianteeman** [comment]
Is this PR outdated - its hard to tell

**Comment by dgt41** [comment]
@brianteeman It seems that it conflicts with the current code. I will update it as soon as we have a decision on #5832

**Comment by dgt41** [comment]
Can this considered RTC? @roland-d ?

**Comment by n9iels** [comment]
@test works fine, but can you change the buttons like @losedk suggested a few messages above?

**Comment by dgt41** [comment]
@n9iels thanks for testing. As I already told @losedk this needs one more PR in order to make the toolbar button compatible with the `footer` option of bootstrap modals. I don’t want to make that here, as it will be to complicate for reviewers o follow the changes

**Comment by n9iels** [comment]
I'm sorry, I didn't saw that :-) If this one is Merged and you have that PR, please post its number here so I can test that one to

@test Succesfull!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4563">issues.joomla.org/joomla-cms/4563</a>.</sub>

**Comment by dgt41** [comment]
@losedk @n9iels In order to have the layout @losedk suggest #6462 needs to be merged!

**Comment by dgt41** [comment]
@roland-d can you RTC this?

**Comment by roland-d** [comment]
@dgt41 This can't go to RTC as it is now, the modal is too wide as mentioned also by @suredweb . I don't think that is an improvement being so wide. Is it a problem to make it less wide?

**Comment by dgt41** [comment]
@roland-d The thing is that right now the default width is 80% of the window (that’s global not only here), so the question is do we keep a global value of 80% as default and override it here or  do we keep a narrow default value (width: 560px; was the bootstrap default) and override in modals that require more estate?

**Comment by roland-d** [comment]
@dgt41 To answer your question, I think we should look at the modals in place. Which ones are more common? If we need the wide ones more than the narrow ones, we should set the wide value as default, otherwise the other way around. Based on that we will know if this modal should be overridden or not.

**Comment by dgt41** [comment]
@roland-d how about this:
screen shot 2015-04-18 at 10 35 41

needs only:
```php
JFactory::getDocument()->addStyleDeclaration(
	"
	div.modal{
		width: 560px;
		margin-left: -280px;
	}
	"
);```
in /administrator/components/com_messages/views/messages/tmpl/default.php
and 
```php
$bar->appendButton('Popup', 'cog', 'COM_MESSAGES_TOOLBAR_MY_SETTINGS', 'index.php?option=com_messages&amp;view=config&amp;tmpl=component', '400px', '250px');
```
in /administrator/components/com_messages/views/messages/view.html.php

**Comment by dgt41** [comment]
We have these modals that need some serious width:
image (media field)
user (user field )
history (version field)
menuselect (menutype field)
module (modal to edit modules in menu)

and I am pretty sure I forget some more…

The only ones that fit better is this messages options and maybe multi language status

**Comment by roland-d** [comment]
@dgt41 So we have more modals that need wide width than narrow width. That explains itself then, we use the wide width as standard and apply local changes where smaller width is needed.

Do you think this can be done by using a class?

**Comment by dgt41** [comment]
@roland-d Ok made some adjustments here:
width 500px
and inline script through API

screen shot 2015-05-19 at 12 22 15

**Comment by roland-d** [comment]
@dgt41 Could you please sync the PR?

**Comment by dgt41** [comment]
@roland-d there you go. #jab15

File: administrator/components/com_messages/views/config/tmpl/default.php [hunk 1]
```diff
@@ -14,59 +14,31 @@
...
-					<div class="control-label">
-						<?php echo $this->form->getLabel('auto_purge'); ?>
-					</div>
-					<div class="controls">
-						<?php echo $this->form->getInput('auto_purge'); ?>
-					</div>
-				</div>
-
-			</div>
+		<?php echo $this->form->getField('lock')->getControlGroup();
```
Comment (wilsonge) [inline administrator/components/com_messages/views/config/tmpl/default.php]: Can you use ```$this->form->renderField('lock');```? https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/form/form.php#L681

Comment (dgt41) [inline administrator/components/com_messages/views/config/tmpl/default.php]: @wilsonge done! Also one fix for the footer buttons (make it work)

**Comment by dgt41** [comment]
@n9iels @suredweb @tairedweb @losedk can anyone give it to test?
This should be the preview for the last iteration:
screenshot 2015-05-30 14 46 05

**Comment by losedk** [comment]
@dgt41 Sure. Give me a minute

**Comment by losedk** [comment]
@dgt41 Issues:

## Responsive:
skaermbillede 2015-05-30 kl 15 31 43

## Hidden tooltip on the first label (possibly not the only place then):
skaermbillede 2015-05-30 kl 15 32 47

## Large button:
skaermbillede 2015-05-30 kl 15 33 10

The success button is inheriting the large width from the toolbar. Perhaps we should change this globally so the button isn't so big in the modals?

## Hathor:
Missing an icon
skaermbillede 2015-05-30 kl 15 34 55

**Comment by dgt41** [comment]
@losedk Thanks, I just fixed the mobile weird behavior. About Hathor there is `<span class="icon-32-cog"></span>` which should load the icon but obviously the icon doesn’t exist and I am not the best bet to design an icon

**Comment by losedk** [comment]
@dgt41 What about?

skaermbillede 2015-05-30 kl 18 10 18

**Comment by dgt41** [comment]
@losedk where can i find that button (i mean which view)?

**Comment by dgt41** [comment]
@losedk I manage to do it! please retest?

**Comment by losedk** [comment]
Could you add it to the stylesheet instead?

**Comment by dgt41** [comment]
Done!

**Comment by losedk** [comment]
@dgt41 All good now!

**Comment by dgt41** [comment]
Moved [here](https://github.com/joomla-extensions/messages/issues/7)

**Comment by zero-24** [comment]
@dgt41 we need the CSS / hathor stuff still in the core :smile:

**Comment by zero-24** [comment]
Can you just remove the messages part and let the css / hathor stuff there?

**Comment by dgt41** [comment]
@zero-24 ok

**Comment by zero-24** [comment]
Thanks @dgt41 I'm going to move this now to RTC :+1:

**Comment by zero-24** [comment]
As it is planned to remove messages with 3.5 (http://developer.joomla.org/news/616-meeting-notes-from-jab-plt-summit-2015.html) We need this here and also https://github.com/joomla-extensions/messages/issues/7 to fix the issue :smile:

**Comment by Kubik-Rubik** [comment]
Thank you @dgt41! Merged in the 3.5-dev branch (https://github.com/joomla/joomla-cms/commit/fa39ddc4d12eb4b4cca1232e92ad29d35a67e380).

**Pull request closed without merging**